libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt

```diff
@@ -168,14 +170,14 @@ abstract class AbstractKotlinCompile<T : CommonCompilerArguments>() : AbstractKo
         get() = (classpath + additionalClasspath)
             .filterTo(LinkedHashSet(), File::exists)
 
-    private val sourceFilesExtensionsSources: MutableList<Iterable<String>> = mutableListOf()
+    private val sourceFilesExtensionsSources = project.objects.listProperty(String::class.java)
```

Comment:
`listProperty` is introduced in Gradle `4.3`, but we support all versions starting from `4.1`

Reply:
Thank you for the quick feedback, @alexey-tsvetkov. I'll see what I can do about it.

Reply:
@bamboo our compatibility policy is to support Gradle versions at least for 1 year, so we can bump the minimal supported version to 4.3 (in theory even to 4.8, but we don't want to deprecate older versions without reason)

Reply:
The initial value for listProperty changed a couple of times (empty list, not set, empty list), so if you want to keep compatibility, you should explicitly set the initial value to an empty list.

Reply:
That's great to hear @alexey-tsvetkov, thanks.

Reply:
Thanks, @big-guy, will do.